hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaStore.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.plugin.etl.batch.commons;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.etl.api.batch.BatchContext;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import com.google.common.base.Joiner;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.hive.hcatalog.data.schema.HCatFieldSchema;
+import org.apache.hive.hcatalog.data.schema.HCatSchema;
+
+import java.util.List;
+
+/**
+ * A class to read/write Hive table schema so that it can be used through various stage of ETL Pipeline.
+ * TODO: This class should be removed once we are able to access information of prepareRun in initialize.
```

Comment:
Yes there is a jira for it and @chtyim is currently working on it. I will add JIRA number todo